### DISCUSSION TITLE
Filter data either by authenticated user or required user_id query param

### DISCUSSION THREAD

**Discussion by pavelozavr** [pocketbase/pocketbase#3414] [Q&A]
Hi, is there a way to filter the data with this logic without middleware or hooks?

So, I want to let the users to get data in two cases:

1. User is authenticated and gets their data by user id
2. A guest user can fetch same data but only by passing user_id param
3. Any user, either authenticated or not, cannot fetch mixed data, or all data from collection

**Comment by ganigeorgiev** [comment]
It is not clear from your description what you are trying to do and what do you mean by _"filter data"_ (which collection?).

I don't understand the last one but in general for the first 2 you can do something like (_assuming that you want to target the users collection_):

```js
id = @request.auth.id ||
id = @request.query.user_id
```
